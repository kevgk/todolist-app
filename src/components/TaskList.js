import React from 'react';
import Task from './Task';
import PropTypes from 'prop-types';

TaskList.propTypes = {
	tasks: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default function TaskList({ tasks, dispatch, setModal }) {
	const taskElements = tasks.map(({ id, name, isChecked }) => (
		<Task
			key={id}
			name={name}
			isChecked={isChecked}
			clickHandler={() => dispatch({ type: 'TOGGLE_TASK', payload: { id } })}
			removeHandler={() => dispatch({ type: 'REMOVE_TASK', payload: { id } })}
			renameHandler={newName =>
				dispatch({
					type: 'RENAME_TASK',
					payload: { id, newName }
				})
			}
			setModal={setModal}
		/>
	));

	const Hint = <div className='noTasksHint'>Let's get productive!</div>;

	return <ul>{tasks.length > 0 ? taskElements : Hint}</ul>;
}
