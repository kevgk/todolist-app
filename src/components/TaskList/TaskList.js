import React, { useContext } from 'react';
import Task from '../Task/Task';
import { TodoContext } from '../../store/todo';
import styles from './TaskList.module.css';

export default function TaskList({ tasks }) {

  const { dispatch } = useContext(TodoContext);

  const taskElements = tasks.map(({ id, name, isChecked }) => (
    <Task
      key={id}
      name={name}
      isChecked={isChecked}
      clickHandler={() => dispatch({ type: 'TOGGLE_TODO', payload: { id } })}
      removeHandler={() => dispatch({ type: 'REMOVE_TODO', payload: { id } })}
      />
    ));

  const Hint = <div className={styles.noTasksHint}>Let's get productive!</div>;

  return (
    <ul>
      { tasks.length > 0 ? taskElements : Hint }
    </ul>
  );
}