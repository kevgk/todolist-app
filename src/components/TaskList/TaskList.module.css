.noTasksHint {
  padding: 0 0 30px;
  margin: auto;
  text-align: center;
}