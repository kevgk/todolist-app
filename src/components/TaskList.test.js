import React from 'react';
import { mount } from 'enzyme';

import TaskList from './TaskList';

it('renders correctly (no tasks)', () => {
	const tasks = [{ id: '0', name: 'Test Task 1', isChecked: false }];
	const wrapper = mount(<TaskList tasks={tasks} />);

	expect(wrapper).toMatchSnapshot();
});

it('renders correctly (1 unchecked task)', () => {
	const tasks = [{ id: '0', name: 'Test Task 1', isChecked: false }];
	const wrapper = mount(<TaskList tasks={tasks} />);

	expect(wrapper).toMatchSnapshot();
});

it('renders correctly (1 checked task)', () => {
	const tasks = [{ id: '0', name: 'Test Task 1', isChecked: true }];
	const wrapper = mount(<TaskList tasks={tasks} />);

	expect(wrapper).toMatchSnapshot();
});
