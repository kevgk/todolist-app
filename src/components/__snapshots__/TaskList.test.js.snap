// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly (1 checked task) 1`] = `
<ul>
  <li
    className="task"
  >
    <button
      className="checkbox"
      onClick={[Function]}
    >
      <span
        aria-label="check task"
        role="img"
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        ✔
      </span>
    </button>
    <div
      className="taskname"
      onClick={[Function]}
    >
      <span>
        Test Task 1
      </span>
    </div>
    <button
      className="removeButton"
      onClick={[Function]}
    >
      x
    </button>
  </li>
</ul>
`;

exports[`renders correctly (1 unchecked task) 1`] = `
<ul>
  <li
    className="task"
  >
    <button
      className="checkbox"
      onClick={[Function]}
    >
      <span
        aria-label="check task"
        role="img"
        style={
          Object {
            "opacity": 0.1,
          }
        }
      >
        ✔
      </span>
    </button>
    <div
      className="taskname"
      onClick={[Function]}
    >
      <span>
        Test Task 1
      </span>
    </div>
  </li>
</ul>
`;

exports[`renders correctly (no tasks) 1`] = `
<ul>
  <li
    className="task"
  >
    <button
      className="checkbox"
      onClick={[Function]}
    >
      <span
        aria-label="check task"
        role="img"
        style={
          Object {
            "opacity": 0.1,
          }
        }
      >
        ✔
      </span>
    </button>
    <div
      className="taskname"
      onClick={[Function]}
    >
      <span>
        Test Task 1
      </span>
    </div>
  </li>
</ul>
`;
